### PR TITLE
update trivy, and unsupport image scanning feature

### LINKED ISSUE #951: Unsupport(Disable) Conrainer Image Scanning
Remove the container image scanning feature using Trivy. I can't maintain it. I think it's better to use Trivy directly to keep up with the latest features.

### PR DESCRIPTION
**Pull request by kotakanbe**
[aquasecurity/trivy](https://github.com/aquasecurity/trivy) , a container image scanner, is evolving rapidly. Vuls used to import Trivy to provide container image scanning, but the maintenance costs of continuing to update Trivy were significant. For users, it's better to use Trivy as is and enjoy the evolution of Trivy, so Vuls will no longer support the image scanning feature.
Library scanning using Trivy will continue to be supported. For example, it would be useful to be able to detect library vulnerabilities by specifying a lock file path in EC2.

To sum it up, here are the following
- Update trivy
- Vuls don't support the image scanning anymore, just use Trivy directly.
- Library scan is continue to be  supported by using trivy.
- Scanning of running containers is continue to be supported.

image

Fixed #951

### CODE REVIEW

File: libmanager/libManager.go [hunk 1]
```diff
@@ -27,7 +45,54 @@ func FillLibrary(r *models.ScanResult) (totalCnt int, err error) {
...
+func downloadDB(appVersion, cacheDir string, quiet, light, skipUpdate bool) error {
+	client := initializeDBClient(cacheDir, quiet)
+	ctx := context.Background()
+	needsUpdate, err := client.NeedsUpdate(appVersion, light, skipUpdate)
+	if err != nil {
+		return xerrors.Errorf("database error: %w", err)
+	}
+
+	if needsUpdate {
+		log.Logger.Info("Need to update DB")
```
Comment (tomoyamachi) [inline libmanager/libManager.go:61]: A `quiet` flag looks suppress any logs.
I expected `-suppress-progress` or so if it only suppresses a progress bar.

File: models/cvecontents.go [hunk 2]
```diff
@@ -313,11 +318,11 @@ var AllCveContetTypes = CveContentTypes{
 	SUSE,
 	DebianSecurityTracker,
 	WPVulnDB,
-	NodeSec,
-	PythonSec,
-	PhpSec,
-	RubySec,
-	RustSec,
+	// NodeSec,
```
Comment (tomoyamachi) [inline models/cvecontents.go:321]: Does not Trivy need here?

File: models/library.go [hunk 3]
```diff
@@ -81,38 +100,22 @@ func (s LibraryScanner) getVulnDetail(tvuln vulnerability.DetectedVulnerability)
...
-			//Cpes          []Cpe             `json:"cpes,omitempty"`
-			//CweIDs        []string          `json:"cweIDs,omitempty"`
-			//Published     time.Time         `json:"published"`
-			//LastModified  time.Time         `json:"lastModified"`
-			//Mitigation    string            `json:"mitigation"` // RedHat API
-			//Optional      map[string]string `json:"optional,omitempty"`
-		}
-		contents[NewCveContentType(source)] = content
+	content := CveContent{
+		Type:          NewCveContentType("trivy"),
```
Comment (tomoyamachi) [inline models/library.go:111]: You can use direct `Trivy`.

File: libmanager/libManager.go [hunk 4]
```diff
@@ -1,10 +1,20 @@
 package libmanager
 
 import (
+	"context"
+
+	db2 "github.com/aquasecurity/trivy-db/pkg/db"
 	"github.com/aquasecurity/trivy/pkg/db"
+	"github.com/aquasecurity/trivy/pkg/github"
+	"github.com/aquasecurity/trivy/pkg/indicator"
 	"github.com/aquasecurity/trivy/pkg/log"
```
Comment (knqyf263) [inline libmanager/libManager.go:10]: Vuls should use your own logger.

File: libmanager/libManager.go [hunk 5]
```diff
@@ -27,7 +45,54 @@ func FillLibrary(r *models.ScanResult) (totalCnt int, err error) {
...
+	client := initializeDBClient(cacheDir, quiet)
+	ctx := context.Background()
+	needsUpdate, err := client.NeedsUpdate(appVersion, light, skipUpdate)
+	if err != nil {
+		return xerrors.Errorf("database error: %w", err)
+	}
+
+	if needsUpdate {
+		log.Logger.Info("Need to update DB")
+		log.Logger.Info("Downloading DB...")
```
Comment (knqyf263) [inline libmanager/libManager.go:62]: ditto

File: models/library.go [hunk 6]
```diff
@@ -3,15 +3,35 @@ package models
...
+	"github.com/aquasecurity/trivy/pkg/types"
 	"github.com/future-architect/vuls/util"
 	"golang.org/x/xerrors"
 
-	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	// "github.com/aquasecurity/go-dep-parser/pkg/types"
 	"github.com/knqyf263/go-version"
 )
 
+// LibraryScanners is a array of LibraryScanner
```
Comment (knqyf263) [inline models/library.go:18]:
```suggestion
// LibraryScanners is an array of LibraryScanner
```

File: models/cvecontents.go (hunk 2 above)
Comment (knqyf263) [reply models/cvecontents.go:321]: You can see it here.
https://github.com/future-architect/vuls/pull/971/files#diff-7d6bddda547d3a8f458448a93dad539dR284

Comment (tomoyamachi) [reply models/cvecontents.go:321]: That line just defined const `Trivy`?
`AllCveContentTypes` does not have `Trivy`.

Maybe we need to set Trivy to `AllCveContentTypes` to show cve titles and so on.
https://github.com/future-architect/vuls/blob/0ff7641471ad115c801de8d5429af4ac237fc04d/models/vulninfos.go#L241

File: libmanager/libManager.go (hunk 1 above)
Comment (kotakanbe) [reply libmanager/libManager.go]: Thanks, add --no-progress flag to tui/report cmdl.

File: models/cvecontents.go (hunk 2 above)
Comment (kotakanbe) [reply models/cvecontents.go:322]: Before the fix, I was downloading Trivy's DB in light mode and using it only for detection.
However, since Trivy's DB has Title, summary, refs, severity and may have information that NVD, OVAL, etc. do not have, I decided to get Trivy's information in full mode and display it in the report.
